planner/core/logical_plan_builder.go

```diff
@@ -2195,23 +2205,45 @@ func (b *PlanBuilder) pushTableHints(hints []*ast.TableOptimizerHint, nodeType n
 	for _, hint := range hints {
 		switch hint.HintName.L {
```

Comment:
What about check `hint.Tables` before the `switch`. Then we can reduce much redundant code.

Reply:
Yes, I have merged these codes. PTAL